src/intersection-observer.js

```diff
@@ -218,10 +223,18 @@ export class IntersectionObserver extends Observable {
   }
 
   /**
-   * Provice a function to clear timeout before set this intersection to null.
+   * Provide a function to clear timeout before set this intersection to null.
    */
   destroy() {
     timer.cancel(this.flushTimeout_);
```

Comment:
I just noticed that we do `this.baseElement_.getViewport().onScroll` above and such. Should we unlisten them as well here?